mapclassify/_classify_API.py

```diff
@@ -0,0 +1,138 @@
+        if using `maximum_breaks` classifier. Deafult =0.
+    initial : int
+        Number of initial solutions to generate or number of runs
+        when using `natural_breaks` or `max_p_classifier`.
+        Default =100.
+        Note: setting initial to 0 will result in the quickest
+        calculation of bins.
+    bins : array, optional
+        (k,1), upper bounds of classes (have to be monotically  
+        increasing) if using `user_defined` classifier.
+        Default =None, Example =[20, max(y)].
+    
+    Returns
+    -------
+    classifier : pysal.mapclassify.classifier instance
+            Object containing bin ids for each observation (.yb),
+            upper bounds of each class (.bins), number of classes (.k)
+            and number of onservations falling in each class (.counts)
+    
+    Note: Supported classifiers include: quantiles, box_plot, euqal_interval,
+        fisher_jenks, headtail_breaks, jenks_caspall, jenks_caspall_forced,
+        max_p_classifier, maximum_breaks, natural_breaks, percentiles, std_mean,
+        user_defined
+    
+    Examples
+    --------
+    Imports
+    
+    >>> from libpysal import examples
+    >>> import geopandas as gpd
+    >>> from mapclassify import classify
+    
+    Load Example Data
+    
+    >>> link_to_data = examples.get_path('columbus.shp')
+    >>> gdf = gpd.read_file(link_to_data)
+    >>> x = gdf['HOVAL'].values
+    
+    Classify values by quantiles
+    
+    >>> quantiles = classify(x, 'quantiles')
+    
+    Classify values by box_plot and set hinge to 2
+    
+    >>> box_plot = classify(x, 'box_plot', hinge=2)
+    
+    """
+    # reformat 
+    scheme_lower = scheme.lower()
+    scheme = scheme_lower.replace('_', '')    
+    
+    # check if scheme is a valid scheme
+    if scheme not in _classifiers:
+        raise ValueError("Invalid scheme. Scheme must be in the"
+                         " set: %r" % _classifiers.keys())
+
+    elif scheme == 'boxplot':
+        classifier = _classifiers[scheme](y, hinge)
+    elif scheme == 'fisherjenkssampled':
+        classifier = _classifiers[scheme](y, k,
+                                          pct_sampled, truncate)
+    elif scheme == 'headtailbreaks':
+        classifier = _classifiers[scheme](y)
+    elif scheme == 'percentiles':
+        classifier = _classifiers[scheme](y, pct)
+    elif scheme == 'stdmean':
+        classifier = _classifiers[scheme](y, multiples)
+    elif scheme == 'jenkscaspallsampled':
+        classifier = _classifiers[scheme](y, k,
+                                          pct_sampled)
+    elif scheme == 'maximumbreaks':
+        classifier = _classifiers[scheme](y, k, mindiff)
+    elif scheme in ['naturalbreaks', 'maxpclassifier']:
+        classifier = _classifiers[scheme](y, k, initial)
+    elif scheme == 'userdefined':
+        classifier = _classifiers[scheme](y, bins)
+    else:
+        classifier = _classifiers[scheme](y, k)
```

Comment:
When does this happen? That is `quantiles` as a default? If so, we should do `def classify(y, scheme="quantiles"...` to make clear what is going on. Might be also good to add a comment to this line saying that it is `quantiles`.

Maybe more readable option would be to specify `quantiles` in `def` as I proposed above and make it also explicit in this condition.

```
if scheme == 'quantiles'

...

elif scheme == 'userdefined'

...
else:
    raise ValueError("Invalid scheme. Scheme must be in the"
                             " set: %r" % _classifiers.keys())
```

It is easier to understand what is going on.

Reply:
Hi @martinfleis it's currently covering FisherJenks, JenksCaspall, JenksCaspallForced, and Quantiles. I am not a big fan of making code longer than necessary, but if you think readability is influenced here I can make all of the statements explicit.

However, in this case I'd recommend using:
```
elif scheme in [...]:
``
Which might make it more readable without introducing too much additional length? 
What do you think about that option?

Reply:
I like your solution.